exercises/collatz-conjecture/example.rs

```diff
@@ -0,0 +1,13 @@
+// return Ok(x) where x is the number of steps required to reach 1
+pub fn collatz(n: u64) -> Result<u64, &'static str> {
+    if n == 0 {
+        return Err("Only positive numbers are allowed");
+    } else if n == 1 {
+        return Ok(0);
+    } else {
+        Ok(1 + match n % 2 {
+            0 => collatz(n / 2).unwrap(),
```

Comment:
Optional: it is a little unfortunate that these `unwrap` are needed. Since we know that non-zero inputs won't result in non-zero outputs, it would be possible to take advantage of it with a non-`pub` function that is `u64 -> u64`.

I will leave it to you whether you wish to actually do that.